pkg/proof/proof.go

```diff
@@ -325,19 +178,19 @@ func GenerateSharesInclusionProof(
 
 		// make sure that the generated root is the same as the eds row root.
 		if !bytes.Equal(rowsRoots[i].Bytes(), tree.Root()) {
-			return types.SharesProof{}, errors.New("eds row root is different than tree root")
+			return types.ShareProof{}, errors.New("eds row root is different than tree root")
 		}
 	}
 
-	return types.SharesProof{
-		RowsProof: types.RowsProof{
-			RowsRoots: rowsRoots,
-			Proofs:    rowsProofs,
-			StartRow:  uint32(startRow),
-			EndRow:    uint32(endRow),
+	return types.ShareProof{
+		RowProof: types.RowProof{
+			RowRoots: rowsRoots,
```

Comment:
[Nitpick] Considering that the field names are changed to singular format, what do you think about changing the argument names to singular as well i.e., `rowsRoots` -> `rowRoots`. Likewise for `sharesProofs` -> `shareProofs`. Btw, I guess that `Roots` and `Proofs` can also be changed to `Root` and `Proof`?.

Reply:
Good catch. I agree with:
- `rowsRoots` -> `rowRoots`
- `sharesProofs` -> `shareProofs`

Since rowRoots and shareProofs are slices, I'm inclined to keep them plural.

Reply:
resolved in https://github.com/celestiaorg/celestia-app/pull/1276/commits/76ee56b0b2d4789ff7a6ac8eddc0a300ef97a191